### PR TITLE
Enable ICU trimming for Invariant=true and PredefinedCulturesOnly unspecified

### PR DESCRIPTION
**Pull request by jkotas**
@RalfKornmannEnvision noticed that ICU initialization is not getting trimmed when Invariant=true and PredefinedCulturesOnly is unspecified (https://gitter.im/dotnet/corert?at=610514ee36b0b97fa2d99aff). The problem is that the Setting class constructor that initializes ICU gets referenced by the PredefinedCulturesOnly property in this case. Moving the PredefinedCulturesOnly out of the Setting class fixes the problem.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs [hunk 1]
```diff
@@ -12,14 +12,13 @@ internal static partial class GlobalizationMode
...
             internal static bool Invariant { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
-            internal static bool PredefinedCulturesOnly { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
         }
 
         // Note: Invariant=true and Invariant=false are substituted at different levels in the ILLink.Substitutions file.
         // This allows for the whole Settings nested class to be trimmed when Invariant=true, and allows for the Settings
         // static cctor (on Unix) to be preserved when Invariant=false.
         internal static bool Invariant => Settings.Invariant;
-        internal static bool PredefinedCulturesOnly => Settings.PredefinedCulturesOnly;
+        internal static bool PredefinedCulturesOnly { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs:21]: Maybe we should rename Settings.Invariant to be something more like InvariantSettings.Enabled to make it less appealing in the future to move other things into it?

**Review by stephentoub — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Review by tarekgh — APPROVED** [review]

**Comment by tarekgh** [comment]
@jkotas this changed made the test InvariantGlobalizationTrue to fail. It looks something used to get trimmed is not trimmed anymore. 

CC @eerhardt

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs [hunk 2]
```diff
@@ -12,13 +12,14 @@ internal static partial class GlobalizationMode
         private static partial class Settings
         {
             internal static bool Invariant { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
+            internal static bool PredefinedCulturesOnly { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", GlobalizationMode.Invariant);
```
Comment (tarekgh) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs:15]: Could you please add comment here telling why we have this property defined in the settings and not defined directly in `GlobalizationMode`

**Review by tarekgh — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs [hunk 3]
```diff
@@ -8,12 +8,15 @@ internal static partial class GlobalizationMode
...
-            /// Load ICU (when not in Invariant mode) in a static cctor to ensure it is loaded early in the process.
-            /// Other places, e.g. CompareInfo.GetSortKey, rely on ICU already being loaded before they are called.
+            /// Load ICU (when not in Invariant mode) in a static cctor to ensure it is loaded as side-effect of
+            /// the GlobalizationMode.Invariant check. Globalization P/Invokes, e.g. in CompareInfo.GetSortKey,
+            /// rely on ICU already being loaded before they are called.
             /// </summary>
             static Settings()
             {
-                if (!Invariant)
+                // Use GlobalizationMode.Invariant to allow ICU initialization to be trimmed when Invariant=false
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs:17]: `when Invariant=false` => `when Invariant=true`

**Pull request merged**